kernel-core/src/main/java/io/zephyr/kernel/core/DefaultPluginContext.java

```diff
@@ -47,4 +75,66 @@ public ModuleTracker createModuleTracker(Predicate<Module> filter) {
     return new AsynchronousModuleThreadTracker(
         kernel, module, (ModuleThread) module.getTaskQueue(), filter);
   }
+
+  @Override
+  public ModuleTracker createModuleTracker(Query<Module> filter) {
+    return createModuleTracker(createFilter(filter));
+  }
+
+  private <T> ExpressionLanguageExtension resolveModuleExpressionLanguageExtension(Query<T> query) {
+    if (log.isLoggable(Level.FINE)) {
+      log.log(Level.FINE, "el.locating.evaluator.modules");
+    }
+    for (val module : kernel.getModuleManager().getModules()) {
+      val lifecycle = module.getLifecycle();
+      val state = lifecycle.getState();
+      if (state.isAtLeast(Lifecycle.State.Active)) {
+        val serviceLoader = resolveModuleExpressionLanguageExtensionInModule(query, module);
+        if (serviceLoader != null) return serviceLoader;
+      }
+    }
+    if (log.isLoggable(Level.FINE)) {
+      log.log(Level.FINE, "el.locating.evaluator.modules.failed");
+    }
+    return null;
+  }
+
+  private <T> ExpressionLanguageExtension resolveModuleExpressionLanguageExtensionInModule(
+      Query<T> query, Module module) {
+    if (log.isLoggable(Level.FINE)) {
```

Comment:
since we call log.isLoggable(Level.FINE) a million times in this class, might be nice to refactor